### PR TITLE
Support volume base tags in InterfaceComputeTags

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Adds support for volume_tags that are base tags with no known type to the
interface tags for mutating compute tags. This cannot be done with non-mutating
compute tags, but since we are moving to mutating compute tags everywhere it
doesn't really matter.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/Domain/Test_InterfaceItems.cpp [hunk 1]
```diff
@@ -90,6 +92,18 @@ struct Negate : db::PrefixTag, db::ComputeTag {
   using argument_tags = tmpl::list<Tag>;
 };
 
+struct NegateDoubleAddInt : db::ComputeTag {
+  static std::string name() noexcept { return "NegateDoubleAddInt"; }
+
+  using return_type = double;
+  static constexpr void function(const gsl::not_null<double*> result,
+                                 const double& x, const int& y) noexcept {
```
Comment (nilsvu) [inline tests/Unit/Domain/Test_InterfaceItems.cpp]: remove `&`s?

**Review by nilsvu — CHANGES_REQUESTED** [review]
LGTM, thanks for making these changes! please squash directly

**Comment by nilsdeppe** [comment]
@nilsleiffischer squashed and rebased on develop. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe  can you please look at this

**Review by wthrowe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**